CoreScriptsRoot/Modules/VR/LaserPointer.lua

```diff
@@ -375,6 +432,76 @@ do --Helper functions
 	function LaserPointer:shouldWalk()
 		return self:isHeadMounted() or self.testWalkToMode
 	end
+
+	function LaserPointer:getRippleOriginPosition()
+		local humanoid = getLocalHumanoid()
+		if not humanoid then return end
+
+		local rootPart = humanoid.Torso
+		if not rootPart then return end
+
+		local hipHeight = humanoid.HipHeight
+		if humanoid.RigType == Enum.HumanoidRigType.R6 then
+			hipHeight = 3
+		end
+
+		return rootPart:GetRenderCFrame().p - Vector3.new(0, hipHeight - 0.01, 0)
+	end
+
+	function LaserPointer:getMaxRippleRadius()
+		local humanoid = getLocalHumanoid()
+		if not humanoid then return 0 end
+
+		return humanoid.WalkSpeed * TELEPORT.RIPPLE_MAX_BUDGET_SECONDS
+	end
+
+	function LaserPointer:calculateRippleRadius()
+		local humanoid = getLocalHumanoid()
+		if not humanoid then return 0 end
+
+		if self.rippleDisabled then
+			return 0
+		end
+
+		local currentTime = tick()
+		local timeSinceRippleStart = currentTime - self.rippleStartTime
+
+		local budget = math.min(timeSinceRippleStart, TELEPORT.RIPPLE_MAX_BUDGET_SECONDS)
+		local studsPerSecond = humanoid.WalkSpeed
+
+		return budget * studsPerSecond
+	end
+
+	function LaserPointer:calculateParabolaRipplePlaneIntersection()
+		-- x = (-b + sqrt(b^2 - 4ac)) / 2a
+		local parabolaOriginCF = self.originPart:GetRenderCFrame()
+		local planeHeight = self:getRippleOriginPosition().Y
+		local a, b, c = self.parabola.A, -self.parabola.B, parabolaOriginCF.p.Y - planeHeight
+
+		if a == 0 then
+			a = 1e-6
```

Comment:
1e-6 seems to be going around a lot here, do we want to put this in some constant?